recipes/z3/all/conanfile.py

```diff
@@ -50,17 +53,26 @@ def configure(self):
             del self.options.fPIC
 
     def requirements(self):
-        self.requires("mpir/3.0.0")
+        if self.options.multiprecision == "mpir":
+            self.requires("mpir/3.0.0")
+        elif self.options.multiprecision == "gmp":
+            self.requires("gmp/6.2.1")
+        elif self.options.multiprecision == "internal":
+            pass
```

Comment:
What does internal means? Will consume some magic library from system? In that case, it's not a good idea, Conan packages should be predictable.

Reply:
It will compile local sources. This is meant to be the compatibility fallback when nothing else is available.

Reply:
I think the recipe should conditionally [add `gmp` to `self.provides` in`configure`](https://docs.conan.io/en/latest/reference/conanfile/attributes.html#provides):
```python
provides = []
def configure(self):
    if self.options.multiprecision == "internal":
        self.provides.append("gmp")
```

If `gmp` and `mpir` also append `gmp`, conan is able to detect collisions.

Reply:
Agreed.
Also, how about having `gmp` the default `multiprecision` setting value?

Reply:
Agreed.
Also, how about having `gmp` the default `multiprecision` setting value ?

Reply:
Okay for me, I just checked and it looks like gmp is available for all platforms.
We can build gmp with MSVC.
In that respect, mpir has much better support.